"X is ambiguous" error is missing a span when the standard library source is not available

This code on the playground
```rust
pub enum Option<T> {
    /// No value
    None,
    /// Some value `T`
    Some(T),
}

pub use Option::*;
```
produces the following error
```
error[E0659]: `Option` is ambiguous
   --> src/lib.rs:8:9
    |
8   | pub use Option::*;
    |         ^^^^^^ ambiguous name
    |
    = note: ambiguous because of multiple potential import sources
note: `Option` could refer to the enum defined here
   --> src/lib.rs:1:1
    |
1   | / pub enum Option<T> {
2   | |     /// No value
3   | |     None,
4   | |     /// Some value `T`
5   | |     Some(T),
6   | | }
    | |_^
    = help: use `crate::Option` to refer to this enum unambiguously
note: `Option` could also refer to the enum defined here
```
Note how the 2nd note is dangling: no span is shown after it says 'could also refer to the enum defined here'.

I was not able to reproduce this locally, so this might also have to do with something special the playground is doing -- Cc @shepmaster

I’d guess it has something to do with the number of crates and their types available by default. 

For example, just typing `Option` by itself says:

``` 
error[E0423]: expected value, found enum `Option`
 --> src/main.rs:2:5
  |
2 |     Option;
  |     ^^^^^^
  |
help: you might have meant to use the following enum variant
  |
2 |     std::option::Option::None;
  |     ~~~~~~~~~~~~~~~~~~~~~~~~~
help: the following enum variant is available
  |
2 |     (std::option::Option::Some(/* fields */));
  |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: consider importing one of these items instead
  |
1 | use serde::de::Unexpected::Option;
  |
1 | use serde_value::Unexpected::Option;
  |
1 | use serde_value::Value::Option;
  |
```

--- a/compiler/rustc_resolve/src/diagnostics.rs
+++ b/compiler/rustc_resolve/src/diagnostics.rs
@@ -1178,7 +1178,7 @@ impl<'a> Resolver<'a> {
 
     fn binding_description(&self, b: &NameBinding<'_>, ident: Ident, from_prelude: bool) -> String {
         let res = b.res();
-        if b.span.is_dummy() {
+        if b.span.is_dummy() || self.session.source_map().span_to_snippet(b.span).is_err() {
             // These already contain the "built-in" prefix or look bad with it.
             let add_built_in =
                 !matches!(b.res(), Res::NonMacroAttr(..) | Res::PrimTy(..) | Res::ToolMod);

